src/pages/project/[id].tsx

```diff
@@ -0,0 +1,132 @@
+import { useRouter } from 'next/router';
+import { useSession } from 'next-auth/react';
+import { useState, useEffect } from 'react';
+import { Project, ProjectSitemap, ProjectDataSchema, ProjectTask } from '~/types';
+
+interface Props {
+  project: Project;
+  sitemap: ProjectSitemap[];
+  dataSchema: ProjectDataSchema[];
+  tasks: ProjectTask[];
+}
+
+const ProjectPage: React.FC<Props> = ({ project, sitemap, dataSchema, tasks }) => {
+  const router = useRouter();
+  const { id } = router.query;
+  const { data: session, status } = useSession();
+  const [projectDetails, setProjectDetails] = useState<Project | null>(null);
+  const [projectSitemap, setProjectSitemap] = useState<ProjectSitemap[]>([]);
+  const [projectDataSchema, setProjectDataSchema] = useState<ProjectDataSchema[]>([]);
+  const [projectTasks, setProjectTasks] = useState<ProjectTask[]>([]);
+
+  useEffect(() => {
+    if (status === 'authenticated' && !session) {
+      router.push('/login');
+    }
+  }, [status, session, router]);
+
+  useEffect(() => {
+    if (id) {
+      // Fetch project details, sitemap, data schema, and tasks using the provided API endpoints and the `id` parameter.
+      // Store the fetched data in the component state using the `useState` hook.
+    }
+  }, [id]);
+
+  const handleApprove = (item: ProjectSitemap | ProjectDataSchema | ProjectTask) => {
+    // Update the corresponding item's `approved` property and send the updated data to the API endpoint to save the changes.
+  };
+
+  const handleReject = (item: ProjectSitemap | ProjectDataSchema | ProjectTask) => {
+    // Update the corresponding item's `approved` property and send the updated data to the API endpoint to save the changes.
+  };
+
+  return (
```

Comment:
Please replace these comments with the actual working code. Check the sourcemap for details about how to call each API endpoint.